app/controllers/admin/listing_shapes_controller.rb

```diff
@@ -50,26 +110,142 @@ def update
       return redirect_to admin_listing_shapes_path
     else
       flash[:error] = t("admin.listing_shapes.edit.update_failure")
-      render("edit", locals: edit_view_locals(shape, get_translations(shape, available_locales()), available_locales()))
+      render("edit", locals: edit_view_locals(shape, processes, available_locales()))
     end
   end
 
 
   private
 
-  def edit_view_locals(shape, translations, available_locs)
+  def valid_template?(template_key, processes)
+    key = template_key.to_sym
+    templates(processes).any? { |tmpl| tmpl[:key] == key }
+  end
+
+  def editable_fields(processes)
+    # TODO Read the processes and define which options are editable
+    {
+      shipping_enabled: true,
+      price_enabled: true,
+      online_payments: true
+    }
+  end
+
+  def templates(transaction_processes)
+    request_process_available = transaction_processes.any? { |tp| tp[:author_is_seller] == false }
+    preauthorize_process_available = transaction_processes.any? { |tp| tp[:process] == :preauthorize }
+
+    template_defaults.reject { |tmpl|
+      tmpl[:key] == :requesting && !request_process_available
+    }
+  end
+
+  def template_defaults
```

Comment:
Actually, if this is moved then `templates(transaction_process)` and `valid_template?(template_key, processes)` be moved too. This feels like good abstraction to have template definitions and methods that operate on those defined in a separate place.